zypp/media/CredentialManager.h

```diff
@@ -37,7 +38,7 @@ namespace zypp
    */
   struct CredManagerOptions
   {
-    CredManagerOptions(const Pathname & rootdir = "");
+    CredManagerOptions(const Pathname & rootdir = ZConfig::instance().repoManagerRoot());
```

Comment:
From a libraries POV the CredentialManager is a standalone usable class, and the 'natural' default value is to use no root (i.e. `/`). I would not change the default here.

If we come to the conclusion that passing the right root in the media backend is too cumbersome, I'd solve it like this:
```
struct MediaCredentialManager : public CredentialManager
{
  MediaCredentialManager() 
  : CredentialManager( CredManagerOptions( ZConfig::instance().repoManagerRoot() ) ) 
  {}
};
```